c++ training/LinkedLists/LinkedLists/DoublyLinkedList.h

```diff
@@ -0,0 +1,323 @@
+	TDLList& operator=(const TDLList<T>& Other)
+	{
+		Clear();
+
+		for (T Item : Other)
+		{
+			AddTail(Item);
+		}
+
+		return *this;
+	}
+	//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+	T& operator[](const int Index)
+	{
+		return GetNodeAt(Index).Data;
+	}
+	//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+	const T& operator[](const int Index) const
+	{
+		return GetNodeAt(Index).Data;
+	}
+	//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+	T& GetHead()
+	{
+		return Head->Data;
+	}
+	//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+	const T& GetHead() const
+	{
+		return Head->Data;
+	}
+	//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+	T& GetTail()
+	{
+		return Tail->Data;
+	}
+	//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+	const T& GetTail() const
+	{
+		return Tail->Data;
+	}
+	//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+	bool IsEmpty() const
+	{
+		return Size == 0;
+	}
+	//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+	int GetSize() const
+	{
+		return Size;
+	}
+	//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+	void AddHead(const T& Data)
+	{
+		if (IsEmpty())
+		{
+			Head = new FNode(Data);
+			Tail = Head;
+
+			Size = 1;
+			return;
+		}
+
+		Head->Previous = new FNode(Data, Head);
+		Head = Head->Previous;
+
+		Size++;
+	}
+	//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+	void AddTail(const T& Data)
+	{
+		if (IsEmpty())
+		{
+			AddHead(Data);
+			return;
+		}
+
+		Tail->Next = new FNode(Data, nullptr, Tail);
```

Comment:
This also happened to me xd at the FNode constructor is better to put (Data, Previous, Next), it makes your life easier

Reply:
Yee, it kinda does feel far more natural to set the previous first